tests/system/HTTP/IncomingRequestDetectingTest.php

```diff
@@ -31,127 +31,150 @@ protected function setUp(): void
 
         $_POST = $_GET = $_SERVER = $_REQUEST = $_ENV = $_COOKIE = $_SESSION = [];
 
-        $origin = 'http://www.example.com/index.php/woot?code=good#pos';
-
+        // The URI object is not used in detectPath().
+        $origin        = 'http://www.example.com/index.php/woot?code=good#pos';
         $this->request = new IncomingRequest(new App(), new URI($origin), null, new UserAgent());
     }
 
     public function testPathDefault()
     {
-        $this->request->uri     = '/index.php/woot?code=good#pos';
+        // /index.php/woot?code=good#pos
         $_SERVER['REQUEST_URI'] = '/index.php/woot';
         $_SERVER['SCRIPT_NAME'] = '/index.php';
-        $expected               = 'woot';
+
+        $expected = 'woot';
         $this->assertSame($expected, $this->request->detectPath());
     }
 
-    public function testPathEmpty()
+    public function testPathDefaultEmpty()
     {
-        $this->request->uri     = '/';
+        // /
         $_SERVER['REQUEST_URI'] = '/';
         $_SERVER['SCRIPT_NAME'] = '/index.php';
-        $expected               = '/';
+
+        $expected = '/';
         $this->assertSame($expected, $this->request->detectPath());
     }
 
     public function testPathRequestURI()
     {
-        $this->request->uri     = '/index.php/woot?code=good#pos';
+        // /index.php/woot?code=good#pos
         $_SERVER['REQUEST_URI'] = '/index.php/woot';
         $_SERVER['SCRIPT_NAME'] = '/index.php';
-        $expected               = 'woot';
+
+        $expected = 'woot';
         $this->assertSame($expected, $this->request->detectPath('REQUEST_URI'));
     }
 
     public function testPathRequestURINested()
     {
-        $this->request->uri     = '/ci/index.php/woot?code=good#pos';
+        // /ci/index.php/woot?code=good#pos
         $_SERVER['REQUEST_URI'] = '/index.php/woot';
         $_SERVER['SCRIPT_NAME'] = '/index.php';
```

Comment:
I don't know what is `Nested` (in the method name), and why
the URI path is `/ci/index.php/woot?code=good#pos` but `$_SERVER['REQUEST_URI']` is `/index.php/woot`.

Reply:
I'm not sure either! Was that my code? It probably just means "in a subdirectory".

Reply:
No, not your code.